great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -423,152 +480,187 @@ def _reconcile_rule(
         expectation_configuration_builder_config: dict
         for (
             expectation_configuration_builder_config
         ) in expectation_configuration_builder_configs:
-            expectation_configuration_builder_name: str = (
-                expectation_configuration_builder_config["expectation_type"]
+            validate_builder_config(
+                builder_config=expectation_configuration_builder_config
             )
-            if (
-                expectation_configuration_builder_name
-                in current_expectation_configuration_builders
-            ):
-                expectation_configuration_builder: ExpectationConfigurationBuilder = (
-                    current_expectation_configuration_builders[
-                        expectation_configuration_builder_name
-                    ]
-                )
-                effective_expectation_configuration_builder_configs.append(
-                    RuleBasedProfiler._reconcile_expectation_configuration_builder(
-                        expectation_configuration_builder=expectation_configuration_builder,
-                        expectation_configuration_builder_config=expectation_configuration_builder_config,
-                    )
-                )
-            else:
-                effective_expectation_configuration_builder_configs.append(
-                    expectation_configuration_builder_config
-                )
 
-        return effective_expectation_configuration_builder_configs
+        effective_expectation_configuration_builder_configs: Dict[str, dict] = {}
 
-    @staticmethod
-    def _reconcile_expectation_configuration_builder(
-        expectation_configuration_builder: ExpectationConfigurationBuilder,
-        expectation_configuration_builder_config: dict,
-    ) -> dict:
-        effective_expectation_configuration_builder_config: dict = {}
-
-        key: str
-        value: Any
-        current_value: Any
-        for key, value in expectation_configuration_builder_config.items():
-            if hasattr(expectation_configuration_builder, f"{key}"):
-                current_value = getattr(expectation_configuration_builder, key)
-                effective_expectation_configuration_builder_config[key] = (
-                    value or current_value
+        current_expectation_configuration_builders: Dict[
+            str, ExpectationConfigurationBuilder
+        ] = rule._get_expectation_configuration_builders_as_dict()
+
+        expectation_configuration_builder_name: str
+        expectation_configuration_builder: ExpectationConfigurationBuilder
+        expectation_configuration_builder_as_dict: dict
+        for (
+            expectation_configuration_builder_name,
+            expectation_configuration_builder,
+        ) in current_expectation_configuration_builders.items():
+            expectation_configuration_builder_as_dict = (
+                expectation_configuration_builder.to_dict()
+            )
```

Comment:
Same note about variable naming - I think it's not the best practice to name the variable based on the type.

Reply:
@cdkini I fixed this (committed).